lib/python/rose/macro.py

```diff
@@ -715,7 +718,18 @@ def validate_config(app_config, meta_config, run_macro_list, modules,
                     else:
                         break
                 if optionals:
-                    res = get_user_values(optionals)
+                    res = {}
+                    if "optional_config_name" in optionals:
+                        res["optional_config_name"] = optional_config_name
+                    for key in set(optionals) & set(optional_values):
+                        res[key] = optional_values[key]
+                    res.update(
+                        get_user_values(
+                            dict(((key, value) for key, value in
+                                  optionals.iteritems() if key not in res))))
```

Comment:
`get_user_values` is called in this awkward way in 2 places in this module. (And is called once in a non-awkward way in `rose.upgrade`.) I wonder if it is worth redesigning the interface for `get_user_values` so it is not as awkward?